processor/resourcedetectionprocessor/internal/azure/aks/aks.go

```diff
@@ -53,6 +55,14 @@ func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	}
 	if d.resourceAttributes.CloudPlatform.Enabled {
 		attrs.PutStr(conventions.AttributeCloudPlatform, conventions.AttributeCloudPlatformAzureAKS)
+
+		if d.resourceAttributes.K8sClusterName.Enabled {
+			m, err := d.provider.Metadata(ctx)
```

Comment:
It doesn't seem like a big deal, but maybe we could actually call this up above so we're not making the request twice

```go
m, err := d.provider.Metadata(ctx)
// If we can't get a response from the metadata endpoint, we're not running in Azure
if err != nil {
  return res, "", nil
}
```